app/src/main/java/io/gnosis/safe/ui/safe/balances/coins/CoinsViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package io.gnosis.safe.ui.safe.balances.coins
+
+import io.gnosis.data.models.Balance
+import io.gnosis.safe.di.Repositories
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import kotlinx.coroutines.flow.collect
+import javax.inject.Inject
+
+class CoinsViewModel
+@Inject constructor(
+    repositories: Repositories,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<CoinsState>(appDispatchers) {
+
+    private val tokenRepositories = repositories.tokenRepository()
+    private val safeRepository = repositories.safeRepository()
+
+    override fun initialState(): CoinsState = CoinsState(loading = false, refreshing = false, viewAction = null)
+
+    init {
+        safeLaunch {
+            safeRepository.activeSafeFlow().collect { load() }
+        }
+    }
+
+    fun load(isRefresh: Boolean = false) {
```

Comment:
you use `isRefresh` here and `refreshing` in a state class. It would be refreshing to have refreshing in both places